lib/conn-pool.js

```diff
@@ -92,7 +127,7 @@ class TCPPool {
     self._pendingConns.push(conn)
     conn.once('close', cleanupPending)
 
-    const peer = Peer.createTCPIncomingPeer(conn)
+    const peer = this.utpServer ? Peer.createUTPIncomingPeer(conn) : Peer.createTCPIncomingPeer(conn)
```

Comment:
Doesn't this assume that if UTP is enabled every incoming connection will be UtP? (from both TCPServer and UTPServer)

Reply:
As you pointed out it can cause confusion, so I've refactored a bit this part 😉